File: Java/problem1423.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int result = 0;
        for (int i = 0; i < k; ++i) {
            result += cardPoints[i];
        }
        int curSum = result;
        for (int i = 0; i < k; ++i) {
            curSum = curSum - cardPoints[k - i - 1] + cardPoints[cardPoints.length - 1 - i];
            result = Math.max(result, curSum);
        }
        return result;
    }
}

File: python/problem1423.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        N = len(cardPoints)
        result = sum(cardPoints[0 : k])
        curSum = result
        for i in range(k):
            curSum += cardPoints[N - 1 - i] - cardPoints[k - 1 - i]
            result = max(result, curSum)
        return result
